tools/src/main/java/org/bitcoinj/tools/BuildCheckpoints.java

```diff
@@ -90,31 +92,50 @@ public static void main(String[] args) throws Exception {
                 throw new RuntimeException("Unreachable.");
         }
 
+        // Configure bitcoinj to fetch only headers, not save them to disk, connect to a local fully synced/validated
+        // node and to save block headers that are on interval boundaries, as long as they are <1 month old.
+        final BlockStore store = new MemoryBlockStore(params);
+        final BlockChain chain = new BlockChain(params, store);
+        final PeerGroup peerGroup = new PeerGroup(params, chain);
+
         final InetAddress ipAddress;
+
+        // DNS discovery can be used for some networks
+        boolean networkHasDnsSeeds = params.getDnsSeeds() != null;
```

Comment:
@schildbach As requested, the method now uses params.getDnsSeeds(). I used a local variable to make the code readable and added comments to all the if/else conditions.